### PR TITLE
dragons_and_princesses_solutions

### PR DESCRIPTION
**Pull request by ikitaichik**

### CODE REVIEW

File: dragon_princess.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3.6
+import yaml
```
Comment (idlv75) [inline dragon_princess.py:2]: How do you ensure that the script can be executed in any environment?

**Review by idlv75 — COMMENTED** [review]
How do you ensure that the script can be executed in any environment?

File: dragon_princess.py (hunk 1 above)
Comment (ikitaichik) [reply dragon_princess.py:2]: Not sure I understand, 

#!/usr/bin/env python3 is the logical path to the interpreter, so it'll work on any filesystem hierarchy standard respecting distro.

if you mean to enforce any environment will have the yaml library,
I'd use something:

try:
    import yaml
except ImportError as e:
    os.system('pip3 install -q pyyaml')
    if importlib.util.find_spec("yaml") is not None:
        print('yaml successfully installed, please rerun this script')
    else:
        print('pip couldn't install yaml')
    os._exit(1)

Comment (ikitaichik) [reply dragon_princess.py:2]: But best practice would be to create a venv with a requirements file and use it:

pip install virtualenv
mkdir New_env
cd New_env/
virtualenv -p <python to use, ex. /usr/bin/python3.6> dragon_princess_env
source dragon_princess_env/bin/activate
pip3 install pyyaml

Create a requirements file so I'll be able to use it in the future 
pip freeze --local > requirements.txt
Then run my script for within that environment so it'll have my dependencies.

Or dockerize it and use that,

FROM python:3
ADD <my_script>
RUN pip install pyyaml  or create a requirements file containing the libraries and use: pip install -r requirements.txt
RUN chmod 644 <my_script>
CMD ["python", "./<my_script>"]

Build the image and then run it

**Comment by idlv75** [comment]
DNP

**Pull request closed without merging**